mule-extensions-api/src/main/java/org/mule/runtime/extension/api/resources/GeneratedResource.java

```diff
@@ -9,28 +9,47 @@
 import org.mule.runtime.extension.api.introspection.ExtensionModel;
 
 /**
- * A resource that supports a
- * {@link ExtensionModel} and which can be generated
- * by the runtime
+ * A resource that is generated in the context of an {@link ExtensionModel}'s
+ * build process
  *
  * @since 1.0
  */
-public interface GeneratedResource
+public final class GeneratedResource
```

Comment:
to make backwards compatibility simpler, we're working under the assumption that all api classes are final until use case shows otherwise